sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuCartesianProductExec.scala

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids
+
+import java.io.{IOException, ObjectInputStream, ObjectOutputStream}
+
+import ai.rapids.cudf.{JCudfSerialization, NvtxColor, NvtxRange}
+import com.nvidia.spark.rapids.{Arm, GpuBindReferences, GpuColumnarBatchSerializer, GpuColumnVector, GpuExec, GpuExpression, GpuSemaphore}
+import com.nvidia.spark.rapids.RapidsPluginImplicits._
+
+import org.apache.spark.{Dependency, NarrowDependency, Partition, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.serializer.Serializer
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
+import org.apache.spark.sql.execution.{BinaryExecNode, ExplainUtils, SparkPlan}
+import org.apache.spark.sql.execution.joins.BuildLeft
+import org.apache.spark.sql.execution.metric.{SQLMetric, SQLMetrics}
+import org.apache.spark.sql.rapids.execution.GpuBroadcastNestedLoopJoinExec
+import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
+import org.apache.spark.util.{CompletionIterator, Utils}
+
+@SerialVersionUID(100L)
+class GpuSerializableBatch(batch: ColumnarBatch)
```

Comment:
Does it make sense to reuse code between this and the serialization code in `GpuBroadcastExchangeExec`?  Seems like there should be some overlap there since both are trying to ship tables around.

Reply:
Broadcast exchange does this in two steps going through the CPU on the driver first.  They are different enough that it is hard to make the reuse clean.